app/controllers/idv/in_person/address_controller.rb

```diff
@@ -5,14 +5,36 @@ class AddressController < ApplicationController
 
       before_action :render_404_if_in_person_residential_address_controller_enabled_not_set
       before_action :confirm_in_person_state_id_step_complete
-      before_action :confirm_in_person_address_step_needed
+      before_action :confirm_in_person_address_step_needed, only: :show
+      before_action :confirm_ssn_step_needed
 
       def show
         analytics.idv_in_person_proofing_address_visited(**analytics_arguments)
 
         render :show, locals: extra_view_variables
       end
 
+      def update
+        flow_session['Idv::Steps::InPerson::AddressStep'] = true
```

Comment:
I think you want to move this to the `if form_result.success?` block

Reply:
we moved this line to success block